src/defaultsettings.cpp

```diff
@@ -26,9 +26,71 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #include "mapgen/mapgen.h" // Mapgen::setDefaultSettings
 #include "util/string.h"
 
+
+/* based on https://github.com/systemd/systemd/blob/7aed43437175623e0f3ae8b071bbc500c13ce893/src/hostname/hostnamed.c#L406
+ * this could be done in future with D-Bus using query:
+ * busctl get-property org.freedesktop.hostname1 /org/freedesktop/hostname1 org.freedesktop.hostname1 Chassis
+ */
+bool detect_touch()
+{
+#if defined(__ANDROID__)
+	return true;
+#else
+	std::string chassis_type;
+#if defined(__linux__) || defined(__FreeBSD__) || defined(__NetBSD__) || defined(__DragonFly__)
```

Comment:
BSDs generally do not have /proc and /sys, no?

Reply:
BSD doesn't have `/sys` but `/proc` can be mounted. I would anyway omit the BSD flavors then.